sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala

```diff
@@ -54,13 +55,21 @@ class ParquetFilterSuite extends QueryTest with ParquetTest with SharedSQLContex
         .select(output.map(e => Column(e)): _*)
         .where(Column(predicate))
 
+      var maybeRelation: Option[BaseRelation] = None
       val maybeAnalyzedPredicate = query.queryExecution.optimizedPlan.collect {
-        case PhysicalOperation(_, filters, LogicalRelation(_: ParquetRelation, _)) => filters
+        case PhysicalOperation(_, filters, LogicalRelation(parquetRelation: ParquetRelation, _)) =>
+          maybeRelation = Some(parquetRelation)
+          filters
       }.flatten.reduceLeftOption(_ && _)
-      assert(maybeAnalyzedPredicate.isDefined)
 
-      val selectedFilters = maybeAnalyzedPredicate.flatMap(DataSourceStrategy.translateFilter)
-      assert(selectedFilters.nonEmpty)
+      assert(maybeRelation.exists(_.isInstanceOf[ParquetRelation]), "Datasource is not Parquet")
```

Comment:
This assertion is redundant since we've already specified `ParquetRelation` in the pattern match.